docker/ubuntu_22.04-dev.dockerfile

```diff
@@ -65,29 +74,50 @@ RUN conda uninstall -y conda-libmamba-solver
 RUN conda config --set solver classic
 RUN conda update -y --all && \
     mamba install -y \
-                cmake \
+                make \
+                gxx_linux-64 \
+                gcc_linux-64 \
                 git \
+                cmake \
+                gfortran \ 
                 libblas \
                 liblapack \
-                hdf5 \
-                setuptools \
+                eigen \
+                "hdf5<1.14" \
+                numpy \
+                scipy \
+                "cython<3" \
+                nose \
                 pytest \
                 pytables \
+                matplotlib \
                 jinja2 \
-                "cython<3" \
+                setuptools \
+                future \
+                progress \
                 && \
     mamba install -y --force-reinstall libsqlite && \
     conda clean -y --all
 RUN mkdir -p `python -m site --user-site`
 
+ENV CC /opt/conda/bin/x86_64-conda_cos6-linux-gnu-gcc
+ENV CXX /opt/conda/bin/x86_64-conda_cos6-linux-gnu-g++
+ENV CPP /opt/conda/bin/x86_64-conda_cos6-linux-gnu-cpp
+
+# put conda on the path
+ENV LD_LIBRARY_PATH /opt/conda/lib:$LD_LIBRARY_PATH
```

Comment:
The error arises because LD_LIBRARY_PATH has been defined to the conda directory, causing issues with the source build. To address this, try unsetting LD_LIBRARY_PATH. However, there's no need to install hdf5 twice for the conda build method.